### PR TITLE
ceph: fix CephMonQuorumAtRisk Alert Query

### LINKED ISSUE #8156: PrometheusRule CephMonQuorumAtRisk rule is wrong
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The CephMonQuorumAtRisk rule is wrong. It alerts in case of a single-mon deployment and alerts too late for five or more mons.

**Expected behavior:**
1 mon: should not alert at all
5 mons: should alert when 3/5 are reachable, alerts at two mons reachable (quorum lost)
7 mons: should alert when 4/7 are alive, alerts at two mons reachable (quorum lost)

**How to reproduce it (minimal and precise):**
Spawn a five mon deployment, kill two mons, no alert is firing.
The alert rule deployed is: 
`count(ceph_mon_quorum_status{job=\"rook-ceph-mgr\"} == 1) <= ((count(ceph_mon_metadata{job=\"rook-ceph-mgr\"}) % 2) + 1)\n` which always evaluates to one or two on the right side regardless of the mon count.
I think it should be:
`count(ceph_mon_quorum_status{job=\"rook-ceph-mgr\"} == 1) <= (floor(count(ceph_mon_metadata{job=\"rook-ceph-mgr\"}) / 2) + 1)\n`.

Also for the case of a single-mon cluster, I think this should not be deployed at all.

**File(s) to submit**:

**Environment**:
- Independent from the environment. 
- Affetcs all rook versions

### PR DESCRIPTION
**Pull request by anmolsachan**
The updated query will work for single mon deployment and will work better for deployments with five or more mons.

Signed-off-by: Anmol Sachan <anmol13694@gmail.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #8156

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [x] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml [hunk 1]
```diff
@@ -349,4 +349,4 @@ spec:
         (ceph_pool_stored_raw * on (pool_id) group_left(name)ceph_pool_metadata) / ((ceph_pool_quota_bytes * on (pool_id) group_left(name)ceph_pool_metadata) > 0) > 0.90
       for: 1m
       labels:
-        severity: critical
\ No newline at end of file
+        severity: critical
```
Comment (parth-gr) [inline cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:352]: plz, Leave an empty line at the end of the file.

Comment (anmolsachan) [reply cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:352]: @parth-gr Thanks for the review. Is it good now?

File: cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml [hunk 2]
```diff
@@ -80,7 +80,7 @@ spec:
         severity_level: error
         storage_type: ceph
       expr: |
-        count(ceph_mon_quorum_status{job="rook-ceph-mgr"} == 1) <= ((count(ceph_mon_metadata{job="rook-ceph-mgr"}) % 2) + 1)
+        count(ceph_mon_quorum_status{job="rook-ceph-mgr"} == 1) <= (floor(count(ceph_mon_metadata{job="rook-ceph-mgr"}) / 2) + 1)
```
Comment (subhamkrai) [inline cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:83]: @travisn @anmolsachan recently, we allowed having even mons too, is this something we need to handle here?

Comment (travisn) [reply cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml:83]: This formula should also work for an even number of mons.

**Review by travisn — APPROVED** [review]

**Pull request merged**